src/GameLogic/DefaultDropGenerator.cs

```diff
@@ -402,7 +402,9 @@ private void AddRandomExcOptions(Item item)
 
     private DropItemGroup? SelectRandomGroup(IEnumerable<DropItemGroup> dropGroups)
     {
-        double lot = this._randomizer.NextDouble();
+        var sum = Math.Max(1.0, dropGroups.Sum(group => group.Chance));
```

Comment:
Is it OK to have sum greater than 1?

It is a case when you have 'rare items' group with chance 0.000001 and it shouldn't drop all the time. Math.Max fix it.